packages/auth/amplify_auth_cognito_dart/lib/src/state/machines/sign_in_state_machine.dart

```diff
@@ -182,47 +179,30 @@ class SignInStateMachine extends StateMachine<SignInEvent, SignInState> {
       _password = password;
     }
 
-    authFlowType = event.authFlow ?? defaultAuthFlowType;
-    if (authFlowType != AuthFlowType.customAuth) {
-      return initiateSrp(event);
+    String expectPassword() {
+      if (password == null) {
+        throw const InvalidParameterException('No password was provided');
+      }
+      return password;
     }
 
-    // If a password is provided, start the SRP flow by including
-    // `CHALLENGE_NAME` in the auth parameters.
-    if (password != null) {
-      final initRequest = await initiateSrp(event);
-      return initRequest.rebuild(
-        (b) => b
-          ..authFlow = AuthFlowType.customAuth
-          ..authParameters.addAll({
-            CognitoConstants.challengeParamChallengeName: 'SRP_A',
-          }),
-      );
+    authFlowType = event.authFlow ?? defaultAuthFlowType;
+    switch (authFlowType) {
+      case AuthFlowType.userSrpAuth:
+        expectPassword();
+        return initiateSrpAuth(event);
+      case AuthFlowType.customAuth:
+        return initiateCustomAuth(event);
+      case AuthFlowType.userPasswordAuth:
+        return initiateUserPasswordAuth(event, expectPassword());
+      case AuthFlowType.refreshToken:
+      case AuthFlowType.refreshTokenAuth:
+      case AuthFlowType.adminNoSrpAuth:
```

Comment:
Why do we have the admin flow types here? These require developer credentials and I don't think would ever be in a client-side app.

Reply:
It's part of the SDK's enum

Reply:
I think it's just the diff formatting. The next line after the break is `throw StateError('Unsupported auth flow')`: https://github.com/aws-amplify/amplify-flutter/pull/1767/files/6add824531df9a3d7fd1336bec88acb25f5e2182#diff-9d1c23e771518da5d10ac1f888bd38c5f865868332b126a16ab6d19fd3ef028fR205